### ISSUE TITLE
[Linux] crashes after connecting to a game no error messages

### ISSUE THREAD

**Issue by Nightshadow1258** [ValveSoftware/Dota-2#1408]
Hi,

I cannot play since yesterday, because the game just closes as soon the loading screen finishes. It does not matter if it is a normal game, a custom game or a local lobby. 

I reported my problem already to the steam community which lead me here.
http://steamcommunity.com/app/570/discussions/0/2860219962095621849/

The Problem is i don't get any error message even if Steam is started from the terminal.
I hope this will be resolved somehow.

I am running Ubuntu 17.10 with the latest drivers on an AMD RX 480

edit tried kernel version 4.14 and 4.13 same problem

**Comment by kisak-valve** [comment]
Helllo @Nightshadow1258, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. Additionally, please add `-con_logfile` to Dota 2's launch options, reproduce the issue, then put `<dota 2 install dir>/game/dota/console.log` in a gist and include a link to it in this issue report.

**Comment by Nightshadow1258** [comment]
Hi,thanks for the replay

here is the link to the System informations:
https://gist.github.com/Nightshadow1258/57b430ec557b7b472aafef54d196a297

here is the link to the console.log after reproducing the crash:
https://gist.github.com/Nightshadow1258/8e4c64f011cad3506d2493e2ae4b188f

**Comment by gdrewb-valve** [comment]
The lack of an error report or spew suggests it's a system issue, such as something in the OpenGL stack.  It could be something else entirely but that's a common reason that Dota has no error information.

Make sure you validate your Dota install from the Steam client.

**Comment by Nightshadow1258** [comment]
I already did a complete reinstall of Dota which did not help sadly.

**Comment by ghost** [comment]
all done today (my only day off)
fresh install of 17.10 with default everything selection.
after install & all system updates done + reboot i do
sudo apt install steam steam-devices -y
connect to steam and install dota, default location
rx480 + ryzen1700
any game, instant crash to desktop as soon as i reach character selection
when i run steam from terminal and instigate the crash, it shows nothing.

was originally on ubuntu 18.04 dailies and i though it was because of that, which is why i fresh installed today.

thank you for your work.

**Comment by ghost** [comment]
work around
sudo apt install mesa-vulkan-drivers vulkan-utils
add -vulkan to launch options and use vulkan instead.

game still stalls (for about 0.4s) at char selection section but it doesn't crash. hopefully this helps others.

**Comment by Veske** [comment]
Can provide a core dump. Happened with` -vulkan` option and without. I was playing a game with Invoker and doing a sun strike on this Phurist Thunderwrath hero and then the game crashed. From now on opening the game will make the game crash in about 15 seconds in.

Some system info:
```

phoronix-test-suite system-info

Phoronix Test Suite v7.6.0
System Information


  PROCESSOR:          Intel Core i5-4570 @ 3.60GHz
    Core Count:       4
    Extensions:       SSE 4.2 + AVX2 + AVX + RDRAND + FSGSBASE
    Cache Size:       6144 KB
    Microcode:        0x23
    Scaling Driver:   intel_pstate powersave

  GRAPHICS:           MSI AMD Radeon RX 470/480/570/580 8192MB
    Display Driver:   modesetting 1.19.1
    Monitor:          BenQ xl2411t
    Screen:           1920x1080

  MOTHERBOARD:        ASUS B85-PLUS
    Memory:           16384MB
    Chipset:          Intel 4th Gen Core DRAM
    Network:          Realtek RTL8111/8168/8411

  DISK:               128GB Samsung SSD 840 + 1000GB Seagate ST1000DM010-2DM1 + 1000GB Seagate ST1000DM003-1CH1
    File-System:      btrfs
    Mount Options:    relatime rw space_cache ssd subvol=/ subvolid=5
    Disk Scheduler:   CFQ

  OPERATING SYSTEM:   Arch rolling
    Kernel:           4.15.5-1-ARCH (x86_64)
    Desktop:          GNOME Shell 3.26.2
    Compiler:         GCC 7.3.0
```

[dota2.log](https://github.com/ValveSoftware/Dota-2/files/1756748/dota2.log)



EDIT: Could be that now when some issue appeared, I would need to clear shade cache? Because before the crash I was able to play 2 games with only 1 crash.

**Comment by gdrewb-valve** [comment]
@Veske, you appear to have a different issue than at least some of the people posting here and you might want to create a separate entry.  Regardless, please include a pastebin link (or whatever you prefer) to your crash dump.  It's impossible to say if it's related to shader cache or not, although it's unlikely if it occurs with both Vulkan and OpenGL.

**Comment by Veske** [comment]
@gdrewb-valve: The issue resolved for me after doing the following steps:

1.  Delete mesa shade cache folder that was located somewhere in `~/.cache`
2.  Delete mesa folder in `~/.cache`
3.  Delete shader caches in Dota 2 folder and steam folder
4.  Verify integrity of game files for Dota 2 

The problem was only solved after step 4, but step 4 did something because there was 1 file that had to be re-downloaded. Could be that some steps played together well and fixed the issue, or it could be that only 4th was needed from the start.  

You still prefer a separate issue in the tracker for this or would you consider it case closed?

**Comment by gdrewb-valve** [comment]
If things work properly after validating your install there's no need for an issue.  If you see problems in the future after validation then you should open a new issue.

**Comment by gdrewb-valve** [comment]
Also, validation might need to be done after every update.

**Comment by Veske** [comment]
> Also, validation might need to be done after every update.

@gdrewb-valve Would make sense to have a small P.S under update notes then. I think I have never heard of that. Good to know tough, could avoid future issues.

**Comment by gdrewb-valve** [comment]
We have it in the submission form here and I just edited the readme to make it more prominent.

**Comment by zpostfacto** [comment]
Man, it would really be great if we could understand why this was necessary.  I mean "just do a validation after every update" is a workaround, but it would be really outstanding if we could understand why the normal patching process isn't working.  @Veske, I don't suppose that you could copy off your whole dota 2 install directory?  And then next time you get the bug, copy off your current tree to a 2nd directory.  And then do the validation.  And then compare the most recent 2 versions and try to understand what file(s) had to be re-downloaded, and what the file looked like in the oldest copy.  Maybe we can try to understand what is causing the file to be in the wrong state in the first place.

That's asking a lot from you to investigate this issue, I know, but I'm hoping that since you're taking the time to file a bug report, maybe you would be willing to help us out.

**Comment by kisak-valve** [comment]
One of the answers [here](https://stackoverflow.com/questions/16085958/scripts-find-the-files-have-been-changed-in-last-24-hours) can help reduce the number of files to compare before and after verifying the game files.

**Comment by Veske** [comment]
@fletcherdvalve Sure, can do that. No guarantees tough on if and when it happens again.

I would imagine that Steam developers could also include a trigger for this verify step which would log the diff on users disk somewhere in case of corrupt file. Could help with some nasty cases.
